test/unit/stateView/document/query/validateQueryFactory.spec.js

```diff
@@ -0,0 +1,1080 @@
+  let findConflictingConditionsStub;
+  let validateQuery;
+
+  beforeEach(function beforeEach() {
+    findConflictingConditionsStub = this.sinon.stub();
+
+    validateQuery = validateQueryFactory(findConflictingConditionsStub);
+  });
+
+  it('should return valid result if empty query is specified', () => {
+    const result = validateQuery({});
+
+    expect(result).to.be.instanceOf(ValidationResult);
+    expect(result.isValid()).to.be.true();
+  });
+
+  notObjectTestCases.forEach((testCase) => {
+    const { type, value } = testCase;
+    it(`should return invalid result if query is a ${type}`, () => {
+      const result = validateQuery(value);
+
+      expect(result).to.be.instanceOf(ValidationResult);
+      expect(result.isValid()).to.be.false();
+    });
+  });
+  it('should return valid result when some valid sample query is passed', () => {
+    findConflictingConditionsStub.returns([]);
+
+    const result = validateQuery({ where: [['a', '>', 1]] });
+
+    expect(result).to.be.instanceOf(ValidationResult);
+    expect(result.isValid()).to.be.true();
+  });
+
+  describe('where', () => {
+    notArrayTestCases.forEach((testCase) => {
+      const { type, value } = testCase;
+      it(`should return invalid result if "where" is not an array, but ${type}`, () => {
+        const result = validateQuery({ where: value });
+
+        expect(result).to.be.instanceOf(ValidationResult);
+        expect(result.isValid()).to.be.false();
+      });
+    });
+
+    it('should return invalid result if "where" is an empty array', () => {
+      const result = validateQuery({ where: [] });
+
+      expect(result).to.be.instanceOf(ValidationResult);
+      expect(result.isValid()).to.be.false();
+    });
+    it('should return invalid result if "where" contains more than 10 conditions', () => {
+      findConflictingConditionsStub.returns([]);
+
+      const result = validateQuery({
+        where: [
+          ['a', '<', 1],
+          ['a', '<', 1],
+          ['a', '<', 1],
+          ['a', '<', 1],
+          ['a', '<', 1],
+          ['a', '<', 1],
+          ['a', '<', 1],
+          ['a', '<', 1],
+          ['a', '<', 1],
+          ['a', '<', 1],
+          ['a', '<', 1],
+        ],
+      });
+
+      expect(result).to.be.instanceOf(ValidationResult);
+      expect(result.isValid()).to.be.false();
+    });
+    it('should return invalid result if "where" contains conflicting conditions', () => {
+      findConflictingConditionsStub.returns([['a', ['<', '>']]]);
+
+      const result = validateQuery({
+        where: [
+          ['a', '<', 1],
+          ['a', '>', 1],
+        ],
+      });
+
+      expect(result).to.be.instanceOf(ValidationResult);
+      expect(result.isValid()).to.be.false();
+      expect(result.errors[0]).to.be.an.instanceOf(ConflictingConditionsError);
+      expect(result.errors[0].message).to.be.equal('Using multiple conditions (<, >) with a single field ("a") is not allowed');
```

Comment:
There are `getField` and `getOperators` methods so you can use them for assertion instead of message.